### DISCUSSION TITLE
Rename file name after uploaded

### DISCUSSION THREAD

**Discussion by pr0grammr** [spatie/laravel-medialibrary#2626] [Q&A] [answered]
Hey guys, 

is it possible to rename the file name after a file was uploaded? 
changing the name is not the problem.. but the file and each conversion needs to be renamed

**Comment by freekmurze** [comment, marked as answer]
This currently isn't possible to do in one go. Either you must:
- write your own code that updates the db / original file / conversion file
- re-add the file to the same model (you can use [`usingFileName`](https://spatie.be/docs/laravel-medialibrary/v9/api/adding-files#usingfilename)) and delete the old media
